### PR TITLE
chore(merge-tree sequence): deprecations related to combining ops

### PR DESCRIPTION
**Pull request by connorskees**
Deprecations for #18096

### CODE REVIEW

File: packages/dds/merge-tree/src/ops.ts [hunk 1]
```diff
@@ -102,6 +102,10 @@ export interface IMergeTreeRemoveMsg extends IMergeTreeDelta {
 	relativePos2?: IRelativePosition;
 }
 
+/**
+ * @deprecated This functionality lacks test coverage and is largely unused. It
```
Comment (Josmithr) [inline packages/dds/merge-tree/src/ops.ts:106]: For all of the newly deprecated items here, what should consumers use instead? We should document that explicitly. If there are no recommended replacements, I would recommend noting that explicitly as well.

Comment (Josmithr) [reply packages/dds/merge-tree/src/ops.ts:106]: Also nit: I'm not sure calling out that the functionality is untested is relevant to consumers. We could probably just omit that part.

File: packages/dds/merge-tree/src/referencePositions.ts [hunk 2]
```diff
@@ -3,7 +3,8 @@
  * Licensed under the MIT License.
  */
 
-// eslint-disable-next-line import/no-deprecated
+/* eslint-disable import/no-deprecated */
```
Comment (Josmithr) [inline packages/dds/merge-tree/src/referencePositions.ts:6]: Nit: disabling this at the file level will prevent us from catching unexpected usages of other deprecated items. As cumbersome as it is, I think we need to disable this rule on a line-by-line basis.

Comment (connorskees) [reply packages/dds/merge-tree/src/referencePositions.ts:6]: I feel decently strongly that a file-wide ignore is the better option here. This disable will be immediately removed in the PR targeting next, and so will be short-lived in main (<2 weeks) and not exist at all in next. Additionally, any PR merging into main would run into this lint during the merge into next. There are >50 removed types and functions in the upcoming release, and adding an eslint ignore at every usage site clutters the code and causes merge conflicts -- unnecessary work in the name of solving an issue that doesn't exist, though I agree in the general case that file-wide ignores are bad.

File: packages/dds/merge-tree/src/ops.ts (hunk 1 above)
Comment (connorskees) [reply packages/dds/merge-tree/src/ops.ts]: I like calling out that the code is untested as it gives some justification and context behind its removal. That said, I agree the fact that there's no replacement could be made a bit more explicit and in rewording the message I removed that bit.

**Review by Abe27342 — APPROVED** [review]

**Pull request merged**